cluster/gce/gci/configure-helper-common.sh

```diff
@@ -1430,7 +1430,112 @@ scrape_configs:
     - targets: ['scale_out_proxy_ip:8404']
 EOF
   sed -i -e "s/scale_out_proxy_ip/$PROXY_RESERVED_IP/g" /tmp/prometheus-metrics.yaml
-  ./prometheus --config.file="/tmp/prometheus-metrics.yaml" --web.listen-address=":9090" --web.enable-admin-api &
+  nohup ./prometheus --config.file="/tmp/prometheus-metrics.yaml" --web.listen-address=":9090" --web.enable-admin-api > prometheus.log 2>&1 &
+
+}
+
+
+function collecting-pprof {
+  local ports=$1
+  local name=$2
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/profile -o ${name}-profile-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/profile -o ${name}-profile-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/heap -o ${name}-heap-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/heap -o ${name}-heap-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/goroutine -o ${name}-goroutine-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/goroutine -o ${name}-goroutine-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/mutex -o ${name}-mutex-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/mutex -o ${name}-mutex-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/block -o ${name}-block-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/block -o ${name}-block-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/threadcreate -o ${name}-threadcreate-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/threadcreate -o ${name}-threadcreate-${CURRENTTIME}.pprof
+}
+
+# Starts collecting profiling files.
+function start-collect-pprof {
+  echo "Start to collect profiling files"
+  mkdir -p /var/log/pprof
+  while true; do
+    cd /var/log/pprof
+    CURRENTTIME=`date +"%Y-%m-%d-%T"`
+    mkdir "${CURRENTTIME}"
+    cd "${CURRENTTIME}"
+
+    COMPONENTS_PORTS="8080"
+    COMPONENTS_NAME="kube-apiserver"
+    echo "Collecting kube-apiserver pprof at ${CURRENTTIME}"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+
+    COMPONENTS_PORTS="2379"
+    COMPONENTS_NAME="etcd"
+    echo "Collecting etcd pprof at ${CURRENTTIME}"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+
+    COMPONENTS_PORTS="10251"
+    COMPONENTS_NAME="kube-scheduler"
+    echo "Collecting kube-scheduler pprof at ${CURRENTTIME}"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+    
+    COMPONENTS_PORTS="10252"
+    COMPONENTS_NAME="kube-controller-manager"
+    echo "Collecting kube-controller-manager pprof"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+
+    COMPONENTS_PORTS="10250"
+    COMPONENTS_NAME="kubelet"
+    echo "Collecting kubelet pprof"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+    sleep 1800
```

Comment:
Here as well.

Reply:
collect profiling data every 30 minutes for debug purpose.